### ISSUE TITLE
setting DEFERRABLE INITIALLY DEFERRED

### ISSUE THREAD

**Issue by elliotstokes** [sequelize/sequelize#2983]
I have some foreign keys on my table and Im using a transaction because of this I need my fk's to be checked when the transaction is committed. 

Is there any way to set DEFERRABLE INITIALLY DEFERRED on the foreign key constraint in the migrations file. I can't find anything in the documentation at all.

**Comment by mickhansen** [comment]
This is not currently supported.
I haven't actually heard of the statement before, what dialects support it?
And it needs to be declared on the foreign key constraint creation statement?

**Comment by elliotstokes** [comment]
I'm using Postgres.

Documentation here

http://www.postgresql.org/docs/9.0/static/sql-createtable.html

**Comment by MarZab** [comment]
I thought I had the same issue, then found out I had a bug where I didn't use the same transaction but only after I did my research on this so here goes:

DEFERRABLE constraints can be set within a transaction block as well; 
http://www.postgresql.org/docs/9.3/static/sql-set-constraints.html

one quick fix is to make a raw SQL query within the same transaction, "SET CONSTRAINTS ALL DEFERRABLE", but i'm having trouble finding the docs for sequelize.query()

**Comment by janmeier** [comment]
It should be as simle as `sequelize.query('SET CONSTRAINTS ALL DEFERRABLE').then`
http://sequelize.readthedocs.org/en/latest/api/sequelize/#querysql-callee-options-replacements-promise

**Comment by shaunc** [comment]
Is there any hook to adjust the sql for foreign key creation depending on backend? This is specific to postgres, but would be quite useful in a postgres context.

**Comment by janmeier** [comment]
@shaunc There is no hook, but a PR that allows setting deferrable when defining constraints in your model would be very welcome

**Comment by sdepold** [comment]
Hey guys. I think I implemented the necessary code here https://github.com/sequelize/sequelize/pull/3702

Feedback is welcome

**Comment by Caerbannog** [comment]
Note to future readers: the syntax that @evangow suggested works in Sequelive v5.

**Comment by QuentinFarizon** [comment]
It's still not working in v6, looking at the code of addConstraintQuery in abstract/query-generator, it does not seem to be possible for now.
